src/test/java/com/microsoft/jenkins/acs/integration/integration-test.sh

```diff
@@ -0,0 +1,232 @@
+            export ACS_TEST_ACR_NAME="$2"
+            shift; shift
+            ;;
+        --skip-clean)
+            export SKIP_CLEAN=true
+            shift
+            ;;
+        --help)
+            show_help
+            ;;
+        *)
+            echo "Unknown parameter '$key'" >&2
+            show_help
+            ;;
+    esac
+done
+
+# allow the service principal client secret to be entered from the terminal
+if [[ -z "$ACS_TEST_CLIENT_SECRET" ]]; then
+    echo -n "Service principal client secret: "
+    read -s ACS_TEST_CLIENT_SECRET
+    export ACS_TEST_CLIENT_SECRET
+fi
+
+# require service principal details
+if [[ -z "$ACS_TEST_SUBSCRIPTION_ID" ]] || [[ -z "$ACS_TEST_CLIENT_ID" ]] || [[ -z "$ACS_TEST_CLIENT_SECRET" ]] || [[ -z "$ACS_TEST_TENANT" ]]; then
+    echo "Service principal details not specified. Terminate" >&2
+    exit -1
+fi
+
+
+# require Azure CLI login
+if ! az account list >/dev/null; then
+    echo "Azure CLI is not logged in" >&2
+    exit -1
+fi
+
+# require docker environment
+if ! which docker >/dev/null; then
+    echo "Docker is not available in the host" >&2
+    exit -1
+fi
+
+# common suffix for the names
+suffix=$(xxd -p -l 4 /dev/urandom)
+
+# construct resource group name
+if [[ -z "$ACS_TEST_RESOURCE_GROUP" ]]; then
+    export ACS_TEST_RESOURCE_GROUP="acs-test-$suffix"
+    # default to SEA
+    if [[ -z "$ACS_TEST_LOCATION" ]]; then
+        export ACS_TEST_LOCATION=SoutheastAsia
+    fi
+else
+    # require resource group location
+    if [[ -z "$ACS_TEST_LOCATION" ]]; then
+        echo "Location is not specified" >&2
+        exit -1
+    fi
+fi
+
+# ACR name
+if [[ -z "$ACS_TEST_ACR_NAME" ]]; then
+    export ACS_TEST_ACR_NAME="acr$suffix"
+fi
+
+# Kubernetes ACS name
+if [[ -z "$ACS_TEST_KUBERNETES_NAME" ]]; then
+    export ACS_TEST_KUBERNETES_NAME="k8s-$suffix"
+fi
+
+# DCOS ACS name
+if [[ -z "$ACS_TEST_DCOS_NAME" ]]; then
+    export ACS_TEST_DCOS_NAME="dcos-$suffix"
+fi
+
+# SWARM ACS name
+if [[ -z "$ACS_TEST_SWARM_NAME" ]]; then
+    export ACS_TEST_SWARM_NAME="swarm-$suffix"
+fi
+
+# SSH
+export ACS_TEST_ADMIN_USER=azureuser
+export ACS_TEST_PRIVATE_KEY_PATH=$(readlink -f ~/.ssh/id_rsa)
+export ACS_TEST_PUBLIC_KEY_PATH=$(readlink -f ~/.ssh/id_rsa.pub)
+
+if [[ ! -f "$ACS_TEST_PUBLIC_KEY_PATH" ]] || [[ ! -f "$ACS_TEST_PRIVATE_KEY_PATH" ]]; then
+    echo "SSH key files are not present in ~/.ssh" >&2
+    exit -1
+fi
+
+export integration_test_script_pid=$$
+
+post_clean_up() {
+    set +x
+    # wait the testing process to finish
+    while ps -p "$integration_test_script_pid" >/dev/null; do
+        sleep 0.5
+    done
+    set -x
+
+    echo "Clean up resource group $ACS_TEST_RESOURCE_GROUP"
+    az group delete --yes --no-wait --name "$ACS_TEST_RESOURCE_GROUP"
+}
+
+group_exists=$(az group exists --name "$ACS_TEST_RESOURCE_GROUP")
+if [[ "false" == "$group_exists" ]]; then
+    az group create --name "$ACS_TEST_RESOURCE_GROUP" --location "$ACS_TEST_LOCATION"
```

Comment:
call `az login` first if user is not logged in?

Reply:
we check for the Azure CLI login status earlier in the script and terminate with error message if it is not logged in.

`az login` is an interactive process. It may be better to put it outside of the automation script.

```bash
# require Azure CLI login
if ! az account list >/dev/null; then
    echo "Azure CLI is not logged in" >&2
    exit -1
fi
```